static/js/components/widgets/WidgetList.js

```diff
@@ -0,0 +1,66 @@
+// @flow
+import React from "react"
+import { SortableContainer } from "react-sortable-hoc"
+
+import WidgetInstance from "./WidgetInstance"
+
+import type { WidgetInstance as WidgetInstanceType } from "../../flow/widgetTypes"
+import type { FormValue } from "../../flow/formTypes"
+
+type Props = {
+  clearForm: () => void,
+  deleteInstance: (widgetInstance: WidgetInstanceType) => void,
+  submitForm: () => void,
+  widgetInstances: Array<WidgetInstanceType>,
+  form: FormValue<Array<WidgetInstanceType>>
+}
+
+export class WidgetList extends React.Component<Props> {
+  render() {
+    const {
+      form,
+      clearForm,
+      submitForm,
+      deleteInstance,
+      widgetInstances
+    } = this.props
+    return (
+      <div className="widget-list">
+        {form ? (
```

Comment:
as far as I can tell, the `form` prop is being used here as an 'editing' flag more or less (it doesn't look like, in this component or in `WidgetInstance` the `form` object is used for anythign except deciding whether or not to show the editing UI).

if that's right, it might be clearer to just pass down an `editing` boolean that does the same thing